### PR TITLE
feat: allow cross-schema query in promql

### PR DESCRIPTION
**Pull request by sunng87**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This patch adds special matcher `__schema__` for our promql parser for selecting a schema where the table sits. This allows user to do cross-schema query in promql.

The patch also does some refactoring in promql parser.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [ ]  This PR does not require documentation updates.

### CODE REVIEW

File: src/promql/src/planner.rs [hunk 1]
```diff
@@ -115,9 +119,16 @@ impl PromPlannerContext {
         self.field_columns = vec![];
         self.tag_columns = vec![];
         self.field_column_matcher = None;
+        self.schema_name = None;
         self.range = None;
     }
 
+    /// Reset table name and schema to empty
+    fn set_empty_table_name(&mut self) {
```
Comment (yuanbohan) [inline src/promql/src/planner.rs:127]: `reset_schema_and_table_name` ?

File: src/promql/src/planner.rs [hunk 2]
```diff
@@ -2014,57 +2049,61 @@ mod test {
...
-        assert!(catalog_list
-            .register_table_sync(RegisterTableRequest {
-                catalog: DEFAULT_CATALOG_NAME.to_string(),
-                schema: DEFAULT_SCHEMA_NAME.to_string(),
-                table_name,
-                table_id: 1024,
-                table,
-            })
-            .is_ok());
+        for (schema_name, table_name) in table_names {
```
Comment (yuanbohan) [inline src/promql/src/planner.rs:2057]: maybe a better variable name (not `table_names`) is needed to include schema meaning?

Comment (sunng87) [reply src/promql/src/planner.rs]: updated to table_name_tuples

**Comment by sunng87** [comment]
Updated

**Review by yuanbohan — APPROVED** [review]

**Comment by sunng87** [comment]
@waynexia PTAL

**Review by waynexia — COMMENTED** [review]
Looking good. Would prefer to also add some sqlness test

**Comment by sunng87** [comment]
Do we have sqlness for promql? I didn't see them so I added tests in tests-integration.

**Review by waynexia — APPROVED** [review]
👍

**Pull request merged**